packages/next/test/fixtures/00-app-dir-ppr/vercel.json

```diff
@@ -339,6 +339,16 @@
       },
       "mustContain": ":{",
       "mustNotContain": "<html"
+    },
+    {
+      "path": "/api/app-headers",
+      "status": 200,
+      "mustContain": "{\"port\":\"80\"}"
```

Comment:
Shouldn't this be 443?

Reply:
No, it's returning 80 for me

Reply:
My assumption is that our live servers have a reverse-proxy set up that is doing TLS at another layer and proxying all internal requests as http; thus, `req.originalRequest.socket.encrypted` is `false`.

Reply:
Seems like we aren't honoring the correct header https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto indicates it should be 443 if we are connecting over HTTPS

Reply:
Okay interesting- well that header also is based off the incoming request check. It was before this change as well. I'll take a closer look at the incoming request object and see if there's a better check we can do.

Reply:
Hmm maybe we need to adjust the `isHttps` check to also check `x-forward-proto` being already set to `https`?

Reply:
Hmm maybe we need to adjust the `isHttps` check to also check `x-forward-proto` being already set to `https`?

Reply:
The one symbol we might be able to tap into is this: 

```
  [Symbol(NextInternalRequestMeta)]: {
    initURL: 'http://localhost:3000/api/pages-headers',
    initQuery: {},
    initProtocol: 'http',
    clonableBody: {
      finalize: [AsyncFunction: finalize],
      cloneBodyStream: [Function: cloneBodyStream]
    }
  }
  ```
  
  I'm going to see what that logs in a deployment now.

Reply:
I cannot get that symbol to log when deployed. Do you have any ideas?

Reply:
Local: <img width="1190" alt="Screenshot 2024-03-13 at 12 14 07" src="https://github.com/vercel/vercel/assets/16144158/c8f6a5cb-b2fa-4a71-8061-b5bcadc019e1">

Deployed: `   [Symbol(NextInternalRequestMeta)]: {}`

Reply:
Fixed by: https://github.com/vercel/next.js/pull/63303